server/verify.go

```diff
@@ -0,0 +1,84 @@
+package server
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/rsa"
+	"crypto/subtle"
+	"fmt"
+	"hash"
+	"sort"
+
+	"github.com/google/go-tpm-tools/proto"
+	"github.com/google/go-tpm-tools/tpm2tools"
+	"github.com/google/go-tpm/tpm2"
+)
+
+// Verify will take in a public key (currently, only RSA is supported), a proto
+// (contains an attestation and a signature), and will check the attestation is
+// valid or not.
+// In addition to the basic RSA signature checking, this function will also check
+// the PCR digest for quoted data, which can insure that the given PCR state is the
+// same as when signing this attestation. This function will return the "Extra Data"
+// appended in the attestation when quoting if passed all verification.
+func Verify(pubkey crypto.PublicKey, quote *proto.Quote, pcrs *proto.Pcrs) ([]byte, error) {
+	sig, err := tpm2.DecodeSignature(bytes.NewBuffer(quote.GetRawSig()))
+	if err != nil {
+		return nil, fmt.Errorf("signature decoding failed: %v", err)
+	}
+	if rsaPubKey, ok := pubkey.(*rsa.PublicKey); ok {
+		if sig.Alg != tpm2.AlgRSASSA && sig.Alg != tpm2.AlgRSAPSS {
```

Comment:
Don't we need different verification methods for SSA vs PSS?

See: https://golang.org/pkg/crypto/rsa/#VerifyPSS

Reply:
Yes we need, I over looked this part..

Reply:
Yes we need, I over looked this part..

Reply:
I only added support for AlgRSASSA for now, since that the scheme used by the default AIK template